libs/blocks/global-navigation/utilities/utilities.js

```diff
@@ -0,0 +1,80 @@
+import { getConfig, localizeLink } from '../../../utils/utils.js';
+
+export function toFragment(htmlStrings, ...values) {
+  const templateStr = htmlStrings.reduce((acc, htmlString, index) => {
+    if (values[index] instanceof HTMLElement) {
+      return `${acc + htmlString}<elem ref="${index}"></elem>`;
+    }
+    return acc + htmlString + (values[index] || '');
+  }, '');
+
+  const fragment = document.createRange().createContextualFragment(templateStr).children[0];
+
+  Array.prototype.map.call(fragment.querySelectorAll('elem'), (replaceable) => {
+    const ref = replaceable.getAttribute('ref');
+    replaceable.replaceWith(values[ref]);
+  });
+
+  return fragment;
+}
+
+export function debounceCallback(callback, time = 200) {
+  if (typeof callback !== 'function') return undefined;
+
+  let timeout = null;
+
+  return (...args) => {
+    clearTimeout(timeout);
+    timeout = setTimeout(() => callback(...args), time);
+  };
+}
+
+// TODO: should we replace these with proper Locale/country service(s)?
+export function getLocale() {
+  return document.documentElement.getAttribute('lang') || 'en-US';
+}
+
+export function getCountry() {
+  return getLocale()?.split('-').pop() || 'US';
+}
+
+// TODO this is just prototyped
+export const getFedsPlaceholderConfig = () => {
+  const { locale, miloLibs } = getConfig();
+  let libOrigin = 'https://adobe.com/libs';
+  if (window.location.origin.includes('localhost')) {
+    libOrigin = `${window.location.origin}/libs`;
+  }
+  if (window.location.origin.includes('.hlx.')) {
+    libOrigin = miloLibs?.replace('hlx.live', 'hlx.page') || 'https://main--milo--adobecom.hlx.page/libs';
```

Comment:
we only want stage pages and never live pages?

Reply:
Good catch, we should change that to properly include stage for stage and live for live - otherwise that might confuse authors that change placeholders and the wrong sheet is fetched.

@overmyheadandbody do you want to batch that with other smaller improvements for this PR?